python/src/lib.rs

```diff
@@ -81,6 +81,22 @@ impl Expr {
         self.0.clone().matches(Some(&value)).map_err(Error::from)
     }
 
+    #[pyo3(signature = (item=None))]
+    fn reduce(&self, item: Option<Bound<'_, PyDict>>) -> Result<Expr> {
+        let value = match item {
+            Some(item) => {
+                let binding: Value = pythonize::depythonize(item.as_ref())?;
+                Some(binding)
+            }
+            None => None,
+        };
```

Comment:
If you like a functional approach, this is equivalent:
```suggestion
        let value = item.map(|item| pythonize::depythonize(&item)).transpose()?;
```

Reply:
The `map` returns an `Option<Result<T>>` and then the transpose converts to `Result<Option<T>>` and then the `?` converts to `Option<T>`